Lib/asyncio/streams.py

```diff
@@ -406,9 +406,12 @@ async def start_tls(self, sslcontext, *,
 
     def __del__(self, warnings=warnings):
         if not self._transport.is_closing():
-            self.close()
-            warnings.warn(f"unclosed {self!r}", ResourceWarning)
-
+            try:
+                self.close()
+            except RuntimeError:
```

Comment:
Catching all `RuntimeError`s always feels scary -- it could mark other, more serious bugs. How about only issuing the new warning if `self._loop.is_closed()`?